lib/pbench/agent/tool_meister.py

```diff
@@ -1324,9 +1341,9 @@ def _create_tar(
         Returns the CompletedProcess object returned by subprocess.run.
         """
 
-        def tar(tar_args: List):
+        def tar(tar_args_: List):
```

Comment:
Trailing underscore? Why?

Reply:
`tar_args` is a variable defined in the outer scope, which is passed into this inner function as a parameter.  One option would be to remove the parameter from this function and just reference the outer scope variable; however, since we do change its value between calls, I think it's better as an explicit parameter.  But, that creates potential confusion, since the reader has to realize that the inner reference is to the parameter and not the outer scope variable.  So, the trailing underscore disambiguates the two (and highlights for the reader that something interesting is going on here).

(The real reason is, my IDE objected, and I couldn't come up with a better name for the parameter, since it really does mean the same thing as the value declared in the outer scope; and, since PEP-8 provides for [using a trailing underscore](https://peps.python.org/pep-0008/#function-and-method-arguments) in a similar situation (see where I did it to `dir`), I decided to do it here, too.)

Reply:
Neither is a Python keyword, so the PEP-8 rule doesn't really apply, and the odd convention immediately makes the reader (me) stop and wonder "what's going on"? Your actual conflict is by no means obvious, and the mystery is counter-productive to understanding. 😦 An explanation (perhaps in a docstring) would be a better solution.

> single_trailing_underscore_: used by convention to avoid conflicts with Python keyword

Reply:
I'm happy to add an explanation, but, if the names aren't different, then the linter complains, so we need _something_ to address that.

My reading of PEP-8 doesn't conflict with what you quote there, but it doesn't seem to be limited to only that case.  They indicate that appending an underscore is preferable to using abbreviated or "corrupted" spelling.  Given that what I'm trying to do here is avoid a name collision with an outer scope, it seemed like the PEP-8 guidance applied.  (But, if you want to offer an alternate spelling which isn't abbreviated or corrupted, we can go that way.)

Peter likes leading underscores, but I don't...and that's already associated with a convention, one which doesn't apply here any better than using a trailing one.

Reply:
I've added the explanation...but it appears in the diffs _after_ this comment...just FYI so you don't miss it.  😉

Reply:
What about this being named `args`?  Since the context of being the `args` of the `tar` method is sufficient to communicate intent while disambiguating.

While we are here, why does the `tar` method inherit `directory` but not `tar_args`?

It would seem that this is a great opportunity to move the `tar` method to a utility function to avoid all these naming problems along with the inheritance issue.

Reply:
> Peter likes leading underscores, but I don't...and that's already associated with a convention, one which doesn't apply here any better than using a trailing one.

I don't like leading *or* trailing underscores, (at best, leading underscores are a hack because Python lacks true `private` properties and methods.) but I like trailing underscores much less because they're even easier to miss and the convention is very obscure.

And I like Peter's suggestion much better. 💯

Reply:
> What about this being named `args`? Since the context of being the `args` of the `tar` method is sufficient to communicate intent while disambiguating.

`args` it is!  (See..._this_ is why we do reviews!  😁)

> While we are here, why does the `tar` method inherit `directory` but not `tar_args`?

Both values _could_ be accessed the same way; however, since `directory` doesn't change between calls but `tar_args` _does_, it seemed clearer to have the latter passed as a parameter.

> It would seem that this is a great opportunity to move the `tar` method to a utility function to avoid all these naming problems along with the inheritance issue.

I think this is where I'm supposed to say, "Perhaps a follow-on commit could correct that? Posted PRs are always welcome!"  😉  Seriously, though, unless there is a need from other code, I don't think moving it from here would be an improvement.

Reply:
This is addressed in #2769.